cmd/fluxd/main.go

```diff
@@ -185,31 +198,61 @@ func main() {
 		k8sManifests = &kubernetes.Manifests{}
 	}
 
-	var reg registry.Registry
+	// Registry components
+	var cache registry.Registry
+	var cacheWarmer registry.Warmer
+	var warmerQueue registry.Queue
 	{
-		var memcacheClient registry.MemcacheClient
+		// Cache
+		var memcacheClient registryMemcache.Client
 		if *memcachedHostname != "" {
-			memcacheClient = registry.NewMemcacheClient(registry.MemcacheConfig{
+			memcacheClient = registryMemcache.NewMemcacheClient(registryMemcache.MemcacheConfig{
 				Host:           *memcachedHostname,
 				Service:        *memcachedService,
 				Timeout:        *memcachedTimeout,
 				UpdateInterval: 1 * time.Minute,
 				Logger:         log.NewContext(logger).With("component", "memcached"),
 			})
-			memcacheClient = registry.InstrumentMemcacheClient(memcacheClient)
+			memcacheClient = registryMemcache.InstrumentMemcacheClient(memcacheClient)
 			defer memcacheClient.Stop()
 		}
 
 		creds, err := registry.CredentialsFromFile(*dockerCredFile)
 		if err != nil {
 			logger.Log("err", err)
 		}
+		cacheLogger := log.NewContext(logger).With("component", "cache")
+		cache = registry.NewRegistry(
+			registry.NewCacheClientFactory(creds, cacheLogger, memcacheClient, *registryCacheExpiry),
+			cacheLogger,
+			*memcachedConnections,
+		)
+		cache = registry.NewInstrumentedRegistry(cache)
+
+		// Remote
 		registryLogger := log.NewContext(logger).With("component", "registry")
-		reg = registry.NewRegistry(
-			registry.NewRemoteClientFactory(creds, registryLogger, memcacheClient, *registryCacheExpiry),
-			registryLogger,
+		remoteFactory := registry.NewRemoteClientFactory(creds, registryLogger, registryMiddleware.RateLimiterConfig{
+			RPS:   *registryRPS,
+			Burst: *registryBurst,
+		})
+
+		// Warmer
+		queueLogger := log.NewContext(logger).With("component", "warmer_queue")
+		warmerQueue = registry.NewQueue(
+			servicesToRepositories(k8s, queueLogger),
+			queueLogger,
+			100*time.Millisecond,
```

Comment:
Is this a mistake? If I understand it correctly this means the warmer is hitting k8s API ten times a second to ask it for all namespaces/deployments/replication controllers/services (and then probably throwing most of that work away because it doesn't enqueue anything into the channel unless it is empty)